src/debug/javaDebugProvider.ts

```diff
@@ -0,0 +1,45 @@
+import * as path from "path";
+import * as vscode from 'vscode';
+
+import { IDebugProvider, IDockerParser, IDockerResolver, PortInfo } from './debugInterfaces';
+import * as debugUtils from "./debugUtils";
+import { shell } from '../shell';
+
+// Use the java debugger extension provided by microsoft team for java debugging.
+const defaultJavaDebuggerExtension = "vscjava.vscode-java-debug";
+
+export class JavaDebugProvider implements IDebugProvider {
+    constructor(readonly dockerResolver: IDockerResolver) {
+    }
+
+    public getDebuggerType(): string {
+        return "java";

```

Comment:
So this is what, the VS Code language ID?

Reply:
Actually it's the debugger configuration type. Follow vscode development doc, every debugger extension needs register a type for itself. Although in most case, the type equals to the language id.

![image](https://user-images.githubusercontent.com/14052197/35960855-47f53b8c-0ce6-11e8-94e4-67013be65e86.png)